error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/MapRules.java

```diff
@@ -43,6 +45,23 @@ V after(Map<K, V> map, T key) {
     }
   }
 
+  /**
+   * Prefer {@link Map#getOrDefault(Object, Object)} over more contrived alternatives. Note that
+   * this method may return null instead of throwing a NPE.
+   */
+  static final class MapGetOrDefault<K, V, T> {
+    @BeforeTemplate
+    V before(Map<K, V> map, T key, V defaultValue) {
+      return requireNonNullElse(map.get(key), defaultValue);
+    }
+
+    @AfterTemplate
+    @Nullable
```

Comment:
It's only `@Nullable` if the input is; in this case we didn't annotate the params as such.